base/linalg/matmul.jl

```diff
@@ -524,10 +525,11 @@ function _generic_matmatmul!{T,S,R}(C::AbstractVecOrMat{R}, tA, tB, A::AbstractV
             if tB == 'N'
                 for i = 1:mA, j = 1:nB
                     if isempty(A) || isempty(B)
-                        Ctmp = zero(R)
+                        z2 = zero(T)*zero(S) + zero(T)*zero(S)
```

Comment:
If A or B is empty, maybe we could just skip the loop entirely and `fill!(C, zero(R))`?